event/event.go

```diff
@@ -2,57 +2,136 @@ package event
 
 import (
 	"encoding/json"
+	"errors"
 	"strings"
 	"time"
 
 	"go.uber.org/zap/zapcore"
 
 	uuid "github.com/satori/go.uuid"
+	"github.com/serverless/event-gateway/internal/zap"
+	validator "gopkg.in/go-playground/validator.v9"
 )
 
-// Event is a default event structure. All data that passes through the Event Gateway is formatted as an Event, based on
-// this schema.
+// Type uniquely identifies an event type.
+type Type string
+
+// TypeInvoke is a special type of event for sync function invocation.
+const TypeInvoke = Type("invoke")
+
+// TypeHTTP is a special type of event for sync http subscriptions.
+const TypeHTTP = Type("http")
+
+const (
+	mimeJSON           = "application/json"
+	mimeFormMultipart  = "multipart/form-data"
+	mimeFormURLEncoded = "application/x-www-form-urlencoded"
+	TransformationVersion = "0.1"
+)
+
+// Event is a default event structure. All data that passes through the Event Gateway
+// is formatted to a format defined CloudEvents v0.1 spec.
 type Event struct {
-	Type       Type        `json:"event"`
-	ID         string      `json:"id"`
-	ReceivedAt uint64      `json:"receivedAt"`
-	Data       interface{} `json:"data"`
-	DataType   string      `json:"dataType"`
+	EventType          Type                   `json:"eventType" validate:"required"`
+	EventTypeVersion   string                 `json:"eventTypeVersion"`
+	CloudEventsVersion string                 `json:"cloudEventsVersion" validate:"required"`
+	Source             string                 `json:"source" validate:"url,required"`
+	EventID            string                 `json:"eventID" validate:"required"`
+	EventTime          time.Time              `json:"eventTime"`
+	SchemaURL          string                 `json:"schemaURL"`
+	ContentType        string                 `json:"contentType"`
+	Extensions         zap.MapStringInterface `json:"extensions"`
+	Data               interface{}            `json:"data"`
 }
 
 // New return new instance of Event.
 func New(eventType Type, mime string, payload interface{}) *Event {
-	return &Event{
-		Type:       eventType,
-		ID:         uuid.NewV4().String(),
-		ReceivedAt: uint64(time.Now().UnixNano() / int64(time.Millisecond)),
-		DataType:   mime,
-		Data:       payload,
+	event := &Event{
+		EventType:          eventType,
+		CloudEventsVersion: "0.1",
+		Source:             "https://serverless.com/event-gateway/#transformationVersion=" + TransformationVersion,
```

Comment:
`https://serverless.com/event-gateway` without trailing `/`

Reply:
The correct address for the page is `https://serverless.com/event-gateway/` with the slash.